contracts/origination/OriginationControllerBase.sol

```diff
@@ -214,6 +213,44 @@ abstract contract OriginationControllerBase is IOriginationControllerBase, EIP71
             );
         }
     }
-}
 
+    /**
+     * @dev Ensure that one counterparty has signed the loan terms, and the other
+     *      has initiated the transaction.
+     *
+     * @param signingCounterparty       The address of the counterparty who signed the terms.
+     * @param callingCounterparty       The address on the other side of the loan as the signingCounterparty.
+     * @param caller                    The address initiating the transaction.
+     * @param signer                    The address recovered from the loan terms signature.
+     * @param sig                       A struct containing the signature data (for checking EIP-1271).
+     * @param sighash                   The hash of the signature payload (used for EIP-1271 check).
+     */
+    // solhint-disable-next-line code-complexity
+    function _validateCounterparties(
```

Comment:
The reason for putting it in the OCBase contract is so there is not duplicate code across two different OriginationControllers. Additionally, the OCBase contract is home to `isSelfOrApproved()` which gets called twice in `_validateCounterparties()` so it seemed to make more sense to have it located in the OCBase contract. Other OriginationControllers that inherit the OCBase that don't need `_validateCounterparties()` should not be affected since it is an internal function.